tests/unit/parallel/algorithms/remove_copy_if.cpp

```diff
@@ -23,8 +23,8 @@ void test_remove_copy_if(ExPolicy const& policy, IteratorTag)
 
     std::vector<int> c(10007);
     std::vector<int> d(c.size());
-    auto middle = boost::begin(c) + c.size()/2;
-    std::iota(boost::begin(c), middle, std::rand());
+    auto middle = boost::begin(c) + std::rand() % (c.size()/2);
+    std::iota(boost::begin(c), middle, static_cast<int>(std::rand() % c.size()));
```

Comment:
Just to make sure I understand the changes: Was the problem that std::iota was occasionally being given too large a value from std::rand()? And is line 26 changed just so that a random value from the 0th position to the 5003rd position is chosen instead of just the 5003rd position?